src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -3330,31 +3330,44 @@ .maxstack  2
       IL_0001:  ldc.i4.0
       IL_0002:  stloc.0
       // sequence point: var y = (i s ...   }).Chain()
-      IL_0003:  ldloc.0
-      IL_0004:  brfalse.s  IL_000e
-      IL_0006:  br.s       IL_0008
-      IL_0008:  ldloc.0
-      IL_0009:  ldc.i4.1
-      IL_000a:  beq.s      IL_0016
-      IL_000c:  br.s       IL_001e
-      IL_000e:  newobj     ""Program..ctor()""
-      IL_0013:  stloc.2
-      IL_0014:  br.s       IL_0026
-      IL_0016:  newobj     ""Program..ctor()""
-      IL_001b:  stloc.2
-      IL_001c:  br.s       IL_0026
-      IL_001e:  newobj     ""Program..ctor()""
-      IL_0023:  stloc.2
-      IL_0024:  br.s       IL_0026
-      IL_0026:  ldloc.2
-      IL_0027:  callvirt   ""Program Program.Chain()""
-      IL_002c:  stloc.1
+      IL_0003:  ldc.i4.1
+      IL_0004:  brtrue.s   IL_0007
+      // sequence point: switch ...         }
+      IL_0006:  nop
+      // sequence point: <hidden>
+      IL_0007:  ldloc.0
+      IL_0008:  brfalse.s  IL_0012
+      IL_000a:  br.s       IL_000c
+      IL_000c:  ldloc.0
+      IL_000d:  ldc.i4.1
+      IL_000e:  beq.s      IL_001a
+      IL_0010:  br.s       IL_0022
+      // sequence point: new Program()
+      IL_0012:  newobj     ""Program..ctor()""
+      IL_0017:  stloc.2
+      IL_0018:  br.s       IL_002a
+      // sequence point: new Program()
+      IL_001a:  newobj     ""Program..ctor()""
+      IL_001f:  stloc.2
+      IL_0020:  br.s       IL_002a
+      // sequence point: new Program()
+      IL_0022:  newobj     ""Program..ctor()""
+      IL_0027:  stloc.2
+      IL_0028:  br.s       IL_002a
+      IL_002a:  ldc.i4.1
+      IL_002b:  brtrue.s   IL_002e
+      // sequence point: var y = (i s ...   }).Chain()
+      IL_002d:  nop
+      // sequence point: <hidden>
```

Comment:
What's going on here? Why do we have a sequence point for the end of the switch and then another hidden sequence point before the call?

Reply:
This first sequence point is used to reset the debugger's view of what the enclosing "statement". The hidden sequence point is to prevent the debugger from stopping here.  The code after the switch expression should appear to be "in" the enclosing statement.

---
In reply to: [401231872](https://github.com/dotnet/roslyn/pull/42711#discussion_r401231872) [](ancestors = 401231872)

Reply:
It'd be useful to document how these sequence points work in https://github.com/dotnet/roslyn/tree/master/docs/compilers